saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/formSection.js

```diff
@@ -0,0 +1,197 @@
+                     .find('input:text, input:hidden, input:password, input:file, select, textarea')
+                     .val('')
+                     .trigger('clear');
+                 
+                 //Select first element on select list
+                 $this
+                     .find('select')
+                     .selectedIndex=0;
+             }
+             
+         });
+         
+         
+         //Hide form section
+         $formSection.slideUp(400, function() {
+             var html = publishInfoFromFormSection(); 
+             
+             if (html !== "") {
+                 $(options.infoSectionId).html(html);
+             }
+         });
+         
+     }
+     
+     /* 
+      * Publishes form information
+      *      
+      * @return string with inputs & selects values   
+      */
+     function publishInfoFromFormSection() {
+         
+         var $formSection = $(options.formSectionId),
+             html = "";
+         
+         $formSection.find("input:text, input:hidden, select").each(function() {
+             var $this = $(this),
+                 fieldDescription = $this.attr("data-field-desc") || "";
+             
+             
+             if ($this.is("input:radio")) {
+                 
+                 var value = "",
+                     id = $this.attr("id");
+             
+                 if (!$this.is(":checked")) {
+                     return;
+                 }
+                 
+                 if (id !== "") {
+                     value += $("label[for=" + id + "]").text();
+                 }
+                 
+                 html += addPhrase(value, fieldDescription, false);
+                 
+             } else if ($this.is("input")) {
+                 
+                 html += addPhrase($this.val(), fieldDescription, true);
+                     
+             } else if ($this.is("select")) {
+                 
+                 var $optionSelected = $this.find("option:selected"); 
+                 
+                 if ($optionSelected.index() > 0) {
+                     
+                     html += addPhrase($optionSelected.text(), fieldDescription, true);
+                 }
+             }
+             
+         });
+         
+         
+         return html;
+         
+         
+         function addPhrase(text, fieldDescription, separator) {
+             var html = ""
+             
+             if (text !== "") {
+
+                 if (separator) {
+                     html += ", ";
+                 }
+                 html += fieldDescription;
+                 html += boldPhrase(text);
+             }
+             
+             return html;
+         }
+         
+         function boldPhrase(phrase) {
+             return "<b>" + phrase + "</b>";
+         }
+     }
+     
+     function showPublishedValues() {
+         var valuesToPublish = publishInfoFromFormSection(); 
+         
+         if (valuesToPublish !== "") {
```

Comment:
Will this situation ever happen? And if so: wouldn't we really like to update the info section in such a case?